tests/contrib/pynamodb/test_pynamodb.py

```diff
@@ -0,0 +1,175 @@
+        assert span.service == "pynamodb"
+        assert span.resource == "DeleteTable Test"
+        assert len(spans) == 1
+        assert_is_measured(span)
+        assert span.span_type == "http"
+        assert span.get_tag("aws.operation") == "DeleteTable"
+        assert span.get_tag("aws.region") == "us-east-1"
+        assert span.get_tag("aws.agent") == "pynamodb"
+        assert span.duration >= 0
+        assert span.error == 0
+
+    @mock_dynamodb
+    def test_scan(self):
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+        self.conn.scan("Test")
+
+        spans = self.get_spans()
+
+        assert spans
+        span = spans[0]
+
+        assert span.name == "pynamodb.command"
+        assert span.service == "pynamodb"
+        assert span.resource == "Scan Test"
+        assert len(spans) == 1
+        assert_is_measured(span)
+        assert span.span_type == "http"
+        assert span.get_tag("aws.operation") == "Scan"
+        assert span.get_tag("aws.region") == "us-east-1"
+        assert span.get_tag("aws.agent") == "pynamodb"
+        assert span.duration >= 0
+        assert span.error == 0
+
+    @mock_dynamodb
+    def test_scan_on_error(self):
+        try:
+            self.conn.scan("OtherTable")
+        except pynamodb.exceptions.ScanError:
+            spans = self.get_spans()
+            assert spans
+            span = spans[0]
+            assert span.name == "pynamodb.command"
+            assert span.service == "pynamodb"
+            assert span.resource == "Scan OtherTable"
+            assert len(spans) == 1
+            assert_is_measured(span)
+            assert span.span_type == "http"
+            assert span.get_tag("aws.operation") == "Scan"
+            assert span.get_tag("aws.region") == "us-east-1"
+            assert span.get_tag("aws.agent") == "pynamodb"
+            assert span.duration >= 0
+            assert span.error == 1
+            assert span.meta["error.type"] != ""
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
+    @mock_dynamodb
+    def test_user_specified_service(self):
+        from ddtrace import config
+
+        assert config.service == "mysvc"
+
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+
+        self.conn.list_tables()
+
+        span = self.get_spans()[0]
+        assert span.service == "pynamodb"
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_PYNAMODB_SERVICE="mypynamodb"))
+    @mock_dynamodb
+    def test_env_user_specified_pynamodb_service(self):
+        dynamodb_backend.create_table("Test", hash_key_attr="content", hash_key_type="S")
+        self.conn.list_tables()
+
+        span = self.get_spans()[0]
+
+        assert span.service == "mypynamodb", span.service
+
+        self.reset()
+
+        # Global config
+        with self.override_config("pynamodb", dict(service="cfg-pynamodb")):
```

Comment:
```suggestion
        with self.override_config("pynamodb", dict(service_name="cfg-pynamodb")):
```
🙂

Reply:
This explains why the test wasn't passing! I made this change and the tests seem to pass now.